src/core/config.py

```diff
@@ -1,18 +1,27 @@
 """Configurations of parameters and initial conditions for a given simulation."""
 
-from typing import Dict, Optional
+from typing import Dict, List
 from dataclasses import dataclass
+from utils.constants import DEFAULT_MODELS
 
-import parameters
+import core.parameters as parameters
+from utils.constants import ModelEnum
 
 
 @dataclass(frozen=True)
 class Config:
-    """Representation of the parameters and initial conditions of the simulation. This module depends on parameters.py. The variation in performance of different runs of the simulation depends on the variation of config."""
-
-    def __init__(self, parameters: parameters.Parameters, initial_conditions: Dict):
+    """Representation of the parameters and initial conditions of the simulation.
+    This module depends on parameters.py.
+    The variation in performance of different runs of the simulation depends on the variation of config."""
+
+    def __init__(
```

Comment:
Are we not initializing our config based on a file? Where are the inputs being specified? I see a `core.config.make_config(config_path)` in main.py, but this function is not defined anywhere, but does take in a file path